### PR TITLE
Move bash utility functions out of preamble

### LINKED ISSUE #2346: Relocate bash functions out of preamble
### What new functionality do you need?

The preamble is getting a bit cluttered as we add more bash functions, so move those out to a separate bash script that is sourced by the preamble.

### What are the requirements for the new functionality?

Move bash functions (except maybe the `postamble` and `set_strict`/`set_trace`) to a separate script.

### Acceptance Criteria

- [ ] Bash functions moved to new separate script (except as noted above)
- [ ] Source new bash functions script in `preamble.sh`
- [ ] Output identical to existing

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
The preamble was accumulating a bunch of utility functions. These functions are now moved to a separate file that is sourced by the preamble. The only functions remaining in the preamble are those related to script control and logging (`set_trace()`, `set_strict()`, `postamble()`).

Resolves #2346

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Cycled test on Orion

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
We might be able to move the loading of the util script to the j-job header and avoid sourcing it anew every script, but more testing would be needed for that.

File: ush/bash_utils.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        fi
+        value=$(echo "${!template}" | envsubst)
+        # shellcheck disable=SC2086
+        declare ${opts} "${com_var}"="${value}"
+        echo "generate_com :: ${com_var}=${value}"
+    done
+    set_trace
+}
+# shellcheck disable=
+declare -xf generate_com
```
Comment (aerorahul) [inline ush/bash_utils.sh]: This is fine here, but I suggest we do the `declare -xf` of all the functions at the end to collect them in one place.  It hopefully helps with code organization and readability.

Comment (WalterKolczynski-NOAA) [reply ush/bash_utils.sh]: Done.

**Comment by WalterKolczynski-NOAA** [comment]
> We might be able to move the loading of the util script to the j-job header and avoid sourcing it anew every script, but more testing would be needed for that.

This actually won't work as-is because the python paths need the python version and they are set before the j-job.

**Review by KateFriedman-NOAA — APPROVED** [review]
LGTM

**Pull request merged**